From akpm@linux-foundation.org  Wed Mar  5 11:40:48 2008
From: Daniel Drake <dsd@gentoo.org>
Date: Tue, 04 Mar 2008 15:25:09 -0800
Subject: USB: hci_usb: another device with buggy SCO support
To: greg@kroah.com
Cc: linux-usb@vger.kernel.org, akpm@linux-foundation.org, dsd@gentoo.org, marcel@holtmann.org
Message-ID: <200803042325.m24NP9pE003205@imap1.linux-foundation.org>


From: Daniel Drake <dsd@gentoo.org>

As reported at:
https://bugs.gentoo.org/show_bug.cgi?id=203196
https://bugs.launchpad.net/ubuntu/+source/linux-source-2.6.22/+bug/152689

This unbranded bluetooth USB dongle causes the following message to
repeatedly appear in the kernel logs:
    hci_scodata_packet: hci0 SCO packet for unknown connection handle 92

This patch solves the issue. I'm not sure of its correctness, it is based
on work by Ossi Berg and Tim Gardner on the Ubuntu bug tracker.

Signed-off-by: Daniel Drake <dsd@gentoo.org>
Cc: Marcel Holtmann <marcel@holtmann.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/bluetooth/hci_usb.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/bluetooth/hci_usb.c
+++ b/drivers/bluetooth/hci_usb.c
@@ -162,6 +162,9 @@ static struct usb_device_id blacklist_id
 	/* Frontline ComProbe Bluetooth Sniffer */
 	{ USB_DEVICE(0x16d3, 0x0002), .driver_info = HCI_SNIFFER },
 
+	/* Generic/no-name device with buggy SCO support */
+	{ USB_DEVICE(0x0e5e, 0x6622), .driver_info = HCI_BROKEN_ISOC },
+
 	{ }	/* Terminating entry */
 };
 
